### PR TITLE
[BugFix] Fix completion_stream_generator return two stops (#2266)

### LINKED ISSUE #2266: openai v1/completions api completion_stream_generator return two stops
When I use the completion api, 
set stream to true
two stops are returned

```
...

{
  "id": "cmpl-3a688b392f324adcb369c11d97d304c7",
  "created": 2421200,
  "model": "qwen-14b-chat-int4",
  "choices": [
    {
      "index": 0,
      "text": "0",
      "logprobs": null,
      "finish_reason": null
    }
  ]
}

{
  "id": "cmpl-3a688b392f324adcb369c11d97d304c7",
  "created": 2421200,
  "model": "qwen-14b-chat-int4",
  "choices": [
    {
      "index": 0,
      "text": "",
      "logprobs": null,
      "finish_reason": "stop".      # first
    }
  ]
}

{
  "id": "cmpl-3a688b392f324adcb369c11d97d304c7",
  "created": 2421200,
  "model": "qwen-14b-chat-int4",
  "choices": [
    {
      "index": 0,
      "text": "",
      "logprobs": null,
      "finish_reason": "stop" # second
    }
  ],
  "usage": {
    "prompt_tokens": 27,
    "total_tokens": 319,
    "completion_tokens": 292
  }
}

```

### PR DESCRIPTION
**Pull request by allenhaozi**
fix #2266

### CODE REVIEW

File: vllm/entrypoints/openai/api_server.py [hunk 1]
```diff
@@ -629,6 +629,8 @@ async def completion_stream_generator() -> AsyncGenerator[str, None]:
                         usage=final_usage,
                     )
                     yield f"data: {response_json}\n\n"
+                else:
+                    yield f"data: {response_json}\n\n"
```
Comment (Yard1) [inline vllm/entrypoints/openai/api_server.py:633]: why not
```suggestion
                yield f"data: {response_json}\n\n"
```

Comment (allenhaozi) [reply vllm/entrypoints/openai/api_server.py]: @Yard1 Thank you for your suggestion. It has been modified

**Comment by WoosukKwon** [comment]
Hi @allenhaozi, thanks for submitting the PR. According to my understanding, neither the current main branch nor this PR exactly implements the OpenAI API's response format. This PR omits the last generated token. Could you please fix this?

* OpenAI API:
```
data: {"id":"cmpl-8d5HgThgqr3rCPyDdnr4luTSlaT1A","object":"text_completion","created":1704325796,"choices":[{"text":" each","index":0,"logprobs":null,"finish_reason":null}],"model":"davinci-002"}

data: {"id":"cmpl-8d5HgThgqr3rCPyDdnr4luTSlaT1A","object":"text_completion","created":1704325796,"choices":[{"text":" neighborhood","index":0,"logprobs":null,"finish_reason":"length"}],"model":"davinci-002"}

data: [DONE]
```

* Current main:
```
data: {"id": "cmpl-3ffa8be79267457f9cfc139696be5905", "created": 64863, "model": "meta-llama/Llama-2-7b-hf", "choices": [{"index": 0, "text": " many", "logprobs": null, "finish_reason": "length"}]}

data: {"id": "cmpl-3ffa8be79267457f9cfc139696be5905", "created": 64863, "model": "meta-llama/Llama-2-7b-hf", "choices": [{"index": 0, "text": "", "logprobs": null, "finish_reason": "length"}], "usage": {"prompt_tokens": 5, "total_tokens": 12, "completion_tokens": 7}}

data: [DONE]
```

* This PR:
```
data: {"id": "cmpl-3473906e66f0494dbf6abf3d6e807222", "created": 65102, "model": "meta-llama/Llama-2-7b-hf", "choices": [{"index": 0, "text": " its", "logprobs": null, "finish_reason": null}]}

data: {"id": "cmpl-3473906e66f0494dbf6abf3d6e807222", "created": 65102, "model": "meta-llama/Llama-2-7b-hf", "choices": [{"index": 0, "text": "", "logprobs": null, "finish_reason": "length"}], "usage": {"prompt_tokens": 5, "total_tokens": 12, "completion_tokens": 7}}

data: [DONE]
```

**Comment by simon-mo** [comment]
Replaced by #3450. Thank you for finding the bug providing the solution initially!

**Pull request closed without merging**